src/core.jl

```diff
@@ -711,4 +711,31 @@ Base.fill!(A::DArray, x) = begin
         @async remotecall_fetch((A,x)->(fill!(localpart(A), x); nothing), p, A, x)
     end
     return A
+end
+
+# new work #
+
+function star(f1,f2,f3,args::DistributedArrays.DArray...)
+        let ds = ds
+            out1 = [remotecall( ()-> f1( DistributedArrays.localpart(ds),2 )[end],p ) for p = workers()]
+
+            out2 = [0; f2(map(fetch,out1))]
+
+            out3 = Future[]
+                    for i = 1:length(workers())
+                        p = workers()[i]
+                        out2i = out2[i]
+                        push!(out3,remotecall(()->f3(out2i,f1( DistributedArrays.localpart(ds),2) ),p))
```

Comment:
`remotecall`s are not scheduled. You probably want to use `remotecall_wait` and a `@async` block. Look at some of the other implementations in the package to see how it works.